src/types/provider.ts

```diff
@@ -140,3 +145,17 @@ export interface TransactionSimulationResponse {
   trace: TransactionTraceResponse;
   fee_estimation: EstimateFeeResponse;
 }
+
+export interface StateUpdateResponse {
+  block_hash: string;
+  new_root: string;
+  old_root: string;
+  state_diff: {
+    storage_diffs: Array<{
```

Comment:
It might make sense to use the RPC format as the default and not the sequencer format since it is planned for the RPC methods to outlast the sequencer ones.